include/drivers/spi.h

```diff
@@ -235,14 +235,6 @@ struct spi_cs_control {
  * @param slave is the slave number from 0 to host controller slave limit.
  * @param cs is a valid pointer on a struct spi_cs_control is CS line is
  *    emulated through a gpio line, or NULL otherwise.
- *
- * @note Only cs_hold and lock_on can be changed between consecutive
- * transceive call. Rest of the attributes are not meant to be tweaked.
- *
- * @warning Most drivers use pointer comparison to determine whether a
- * passed configuration is different from one used in a previous
- * transaction.  Changes to fields in the structure may not be
- * detected.
```

Comment:
I don't understand why this warning is removed. It looks like nothing changed in drivers behavior in this regard (`spi_context_configured()` still uses pointer comparison).

Reply:
Yes but there cannot be any change within the struct spi_config, this warning was meant to notice about the fact that changing cs_hold bit could not be detected. This is an obsolete warning altogether now, struct spi_config is meant to be const and nothing inside is going to change at runtime.

Reply:
> this warning was meant to notice about the fact that changing cs_hold bit could not be detected.

Not exactly. It was introduced in https://github.com/zephyrproject-rtos/zephyr/commit/7a8a4c9b3e02356cedc22936b2726bbaf5d2be30 to warn users that the same `spi_config` structure cannot be re-used when the transfer configuration needs to be changed, to prevent issues like #26319.

> struct spi_config is meant to be const and nothing inside is going to change at runtime.

How are the users supposed to know this? If this warning is to be removed, such expectation needs to be specified in some other way in the API.

Reply:
I am aware that there are also those notes "Pointer-comparison may be used to detect changes from previous operations" in the `config` parameter descriptions. But this warning is more visible, and as it is still valid, I think it should stay.

Reply:
I agree.

Reply:
As you want, but in the end struct spi_config are/will be meant to all be const, so everything will be set at built time. Obviously no fields are going to change there.